### PR TITLE
EOS-19444: Add new component - "base_utils" for the Foundation

### PR DESCRIPTION
**Pull request by 83bhp**
Added new deployable component - corts_py_utils under base_utils
1. Add new group state - base_utils
2. Add provisoiner component for base_utils - base_utils.cortx_py_utils
2. Make corresponding changes in Jenkins file and provisioner cli code

Signed-off-by: Pritam Bhavsar <pritam.bhavsar@seagate.com>

### CODE REVIEW

File: api/python/provisioner/commands/deploy.py [hunk 1]
```diff
@@ -61,6 +61,9 @@
         "system.logrotate",
         "system.chrony"
     ],
+    base_utils=[
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy.py:64]: Please change this to utils...

File: api/python/provisioner/commands/deploy.py [hunk 2]
```diff
@@ -61,6 +61,9 @@
...
+        "cotx_utils"
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy.py:65]: Please change this to cortx_utils or foundation...

File: api/python/provisioner/commands/deploy.py [hunk 3]
```diff
@@ -382,6 +385,7 @@ def run(self, **kwargs):  # noqa: C901 FIXME
 
             self._rescan_scsi_bus()
             self._run_states('system', run_args)
+            self._run_states('base_utils', run_args)
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy.py:388]: Please change this to utils...

File: api/python/provisioner/commands/deploy.py [hunk 4]
```diff
@@ -398,6 +402,10 @@ def run(self, **kwargs):  # noqa: C901 FIXME
                 logger.info("Deploying the prereq states")
                 self._run_states('prereq', run_args)
 
+            if 'base_utils' in run_args.states:
+                logger.info("Deploying the base_utils states")
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy.py:406]: Deploying the utils states

File: api/python/provisioner/commands/deploy_jbod.py [hunk 5]
```diff
@@ -46,6 +46,9 @@
         'system.logrotate',
         'system.chrony'
     ],
+    base_utils=[
+        "cotx_utils"
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_jbod.py:50]: Please change this to cortx_utils or foundation..

File: api/python/provisioner/commands/deploy_jbod.py [hunk 6]
```diff
@@ -130,6 +133,7 @@ def run(self, **kwargs):
             self._run_states('system', run_args)
             self._encrypt_pillar()
             self._run_states('prereq', run_args)
+            self._run_states('base_utils', run_args)
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_jbod.py:136]: Please change this to utils...

File: api/python/provisioner/commands/deploy_jbod.py [hunk 7]
```diff
@@ -46,6 +46,9 @@
         'system.logrotate',
         'system.chrony'
     ],
+    base_utils=[
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_jbod.py:49]: Please change this to utils...

File: api/python/provisioner/commands/deploy_vm.py [hunk 8]
```diff
@@ -56,6 +56,9 @@
         "system.logrotate",
         "system.chrony"
     ],
+    base_utils=[
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_vm.py:59]: Please change this to utils...

File: api/python/provisioner/commands/deploy_vm.py [hunk 9]
```diff
@@ -218,6 +222,10 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Deploying the system states")
                 self._run_states('system', run_args)
 
+            if 'base_utils' in run_args.states:
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_vm.py:225]: Please change this to utils...

File: api/python/provisioner/commands/deploy_vm.py [hunk 10]
```diff
@@ -56,6 +56,9 @@
         "system.logrotate",
         "system.chrony"
     ],
+    base_utils=[
+        "cotx_utils"
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_vm.py:60]: Please change this to cortx_utils or foundation..

File: api/python/provisioner/commands/deploy_vm.py [hunk 11]
```diff
@@ -205,6 +208,7 @@ def run(self, **kwargs):  # noqa: C901
 
         if run_args.states is None:  # all states
             self._run_states('system', run_args)
+            self._run_states('base_utils', run_args)
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_vm.py:211]: Please change this to utils...

File: api/python/provisioner/commands/deploy_vm.py [hunk 12]
```diff
@@ -218,6 +222,10 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Deploying the system states")
                 self._run_states('system', run_args)
 
+            if 'base_utils' in run_args.states:
+                logger.info("Deploying the base_utils states")
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/deploy_vm.py:226]: Please change this to utils...

File: api/python/provisioner/commands/destroy.py [hunk 13]
```diff
@@ -63,6 +63,9 @@
         "s3server.teardown",
         "motr.teardown"
     ],
+    base_utils=[
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/destroy.py:66]: Please change this to utils...

File: api/python/provisioner/commands/destroy.py [hunk 14]
```diff
@@ -63,6 +63,9 @@
...
+        "cotx_utils.teardown"
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/destroy.py:67]: Please change this to cortx_utils or foundation...

File: api/python/provisioner/commands/destroy.py [hunk 15]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Teardown the system states")
                 self._run_states('system', run_args)
 
+            if 'base_utils' in run_args.states:
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/destroy.py:311]: Please change this to utils...

File: api/python/provisioner/commands/destroy.py [hunk 16]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
...
+                logger.info("Teardown Provisioner base_utils states")
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/destroy.py:312]: Please change this to utils...

File: api/python/provisioner/commands/destroy.py [hunk 17]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
...
+                self._run_states('base_utils', run_args)
```
Comment (nitin-seagate) [inline api/python/provisioner/commands/destroy.py:313]: Please change this to utils...

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 18]
```diff
@@ -199,6 +199,11 @@ node {
                 echo "Successfully deployed system states!"
             }
 
+            stage("Base_utils Software Deployment") {
```
Comment (nitin-seagate) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:202]: Please change this to utils...

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 19]
```diff
@@ -199,6 +199,11 @@ node {
...
+                sshCommand remote: remotes[0], command: "provisioner deploy_vm --states base_utils --setup-type 3_node"
```
Comment (nitin-seagate) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:203]: Please change this to utils...

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 20]
```diff
@@ -199,6 +199,11 @@ node {
...
+                echo "Successfully deployed base_utils states!"
```
Comment (nitin-seagate) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:204]: Please change this to utils...

File: devops/ci/multi_node_deploy/3node_destroy.jenkinsfile [hunk 21]
```diff
@@ -88,6 +88,11 @@ node {
             echo "Successful teardown: 3rd-Party"
         }
 
+        stage("Teardown: base_utils") {
+            sshCommand remote: remotes[0], command: "provisioner deploy_vm --states base_utils --setup-type 3_node"
+            echo "Successful teardown: base_utils"
```
Comment (nitin-seagate) [inline devops/ci/multi_node_deploy/3node_destroy.jenkinsfile:93]: Please change this to utils...

File: devops/ci/single_node_deploy/deploy.jenkinsfile [hunk 22]
```diff
@@ -157,6 +157,11 @@ node {
                 echo "Successfully deployed system states!"
             }
             
+            stage("base_utils Setup") {
```
Comment (nitin-seagate) [inline devops/ci/single_node_deploy/deploy.jenkinsfile:160]: Please change this to utils...

File: devops/ci/single_node_deploy/deploy_iso.jenkinsfile [hunk 23]
```diff
@@ -152,6 +152,13 @@ node {
                 echo "Successfully deployed system states!"
             }
             
+            stage("Base_utils Setup") {
```
Comment (nitin-seagate) [inline devops/ci/single_node_deploy/deploy_iso.jenkinsfile:155]: Please change this to utils...

File: srv/components/cortx_utils/config.sls [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+
+Stage - cortx-utils post_install:
+#/opt/seagate/cortx/utils/bin/utils_setup post_install --config yaml:///tmp/utils.post_install.tmpl.3-node
+  cmd.run:
+    - name: __slot__:salt:setup_conf.conf_cmd('/opt/seagate/cortx/utils/conf/setup.yaml', 'utils:post_install')
+    - failhard: True
+
+Stage - cortx-utils config:
+#/opt/seagate/cortx/utils/bin/utils_setup config --config yaml:///tmp/utils.config.tmpl.3-node  cmd.run:
+    - name: __slot__:salt:setup_conf.conf_cmd('/opt/seagate/cortx/utils/conf/setup.yaml', 'utils:config')
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/config.sls:26]: Do we not need cmd.run?

File: srv/components/cortx_utils/init.sls [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+include:
+  - .install
+  - .config
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/init.sls:20]: Add new line

File: srv/components/cortx_utils/install.sls [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Install - cortx-py-utils:
+  pkg.installed:
+    - pkgs:
+      - cortx-py-utils: latest
+    - refresh: True
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/install.sls:22]: Add new line

File: srv/components/cortx_utils/teardown.sls [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+    - name: __slot__:salt:setup_conf.conf_cmd('/opt/seagate/cortx/utils/conf/setup.yaml', 'utils:reset')
+
+Stage - Cleanup cortx-py-utils:
+  cmd.run:
+    - name: __slot__:salt:setup_conf.conf_cmd('/opt/seagate/cortx/utils/conf/setup.yaml', 'utils:cleanup')
+
+Remove cortx-py-utils package:
+  pkg.purged:
+    - pkgs:
+      - cortx-py-utils
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/teardown.sls:29]: Add new line

File: srv/components/cortx_utils/config.sls [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-utils post_install:
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/config.sls:18]: Something seems to be wrong, can we restructure properly...

**Review by nitin-seagate — CHANGES_REQUESTED** [review]
Something seems to be wrong, can we restructure properly...

File: api/python/provisioner/commands/deploy.py [hunk 29]
```diff
@@ -61,6 +61,9 @@
         "system.logrotate",
         "system.chrony"
     ],
+    utils=[
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:64]: Rename utils to foundation. (Let's finalize this as foundation)

File: api/python/provisioner/commands/deploy.py [hunk 30]
```diff
@@ -382,6 +385,7 @@ def run(self, **kwargs):  # noqa: C901 FIXME
 
             self._rescan_scsi_bus()
             self._run_states('system', run_args)
+            self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:388]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy.py [hunk 31]
```diff
@@ -398,6 +402,10 @@ def run(self, **kwargs):  # noqa: C901 FIXME
                 logger.info("Deploying the prereq states")
                 self._run_states('prereq', run_args)
 
+            if 'utils' in run_args.states:
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:405]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_jbod.py [hunk 32]
```diff
@@ -46,6 +46,9 @@
         'system.logrotate',
         'system.chrony'
     ],
+    utils=[
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_jbod.py:49]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy.py [hunk 33]
```diff
@@ -398,6 +402,10 @@ def run(self, **kwargs):  # noqa: C901 FIXME
...
+                logger.info("Deploying the utils states")
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:406]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy.py [hunk 34]
```diff
@@ -398,6 +402,10 @@ def run(self, **kwargs):  # noqa: C901 FIXME
...
+                self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/deploy.py:407]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_jbod.py [hunk 35]
```diff
@@ -130,6 +133,7 @@ def run(self, **kwargs):
             self._run_states('system', run_args)
             self._encrypt_pillar()
             self._run_states('prereq', run_args)
+            self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_jbod.py:136]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_vm.py [hunk 36]
```diff
@@ -56,6 +56,9 @@
         "system.logrotate",
         "system.chrony"
     ],
+    utils=[
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_vm.py:59]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_vm.py [hunk 37]
```diff
@@ -205,6 +208,7 @@ def run(self, **kwargs):  # noqa: C901
 
         if run_args.states is None:  # all states
             self._run_states('system', run_args)
+            self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_vm.py:211]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_vm.py [hunk 38]
```diff
@@ -218,6 +222,10 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Deploying the system states")
                 self._run_states('system', run_args)
 
+            if 'utils' in run_args.states:
+                logger.info("Deploying the utils states")
+                self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_vm.py:227]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_vm.py [hunk 39]
```diff
@@ -218,6 +222,10 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Deploying the system states")
                 self._run_states('system', run_args)
 
+            if 'utils' in run_args.states:
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_vm.py:225]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_vm.py [hunk 40]
```diff
@@ -218,6 +222,10 @@ def run(self, **kwargs):  # noqa: C901
...
+                logger.info("Deploying the utils states")
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_vm.py:226]: `utils` to `foundation`

File: api/python/provisioner/commands/destroy.py [hunk 41]
```diff
@@ -63,6 +63,9 @@
         "s3server.teardown",
         "motr.teardown"
     ],
+    utils=[
```
Comment (ypise) [inline api/python/provisioner/commands/destroy.py:66]: `utils` to `foundation`

File: api/python/provisioner/commands/destroy.py [hunk 42]
```diff
@@ -291,6 +294,7 @@ def run(self, **kwargs):  # noqa: C901
             self._run_states('controlpath', run_args)
             self._run_states('iopath', run_args)
             self._run_states('prereq', run_args)
+            self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/destroy.py:297]: `utils` to `foundation`

File: api/python/provisioner/commands/destroy.py [hunk 43]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Teardown the system states")
                 self._run_states('system', run_args)
 
+            if 'utils' in run_args.states:
```
Comment (ypise) [inline api/python/provisioner/commands/destroy.py:311]: `utils` to `foundation`

File: api/python/provisioner/commands/destroy.py [hunk 44]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
...
+                logger.info("Teardown Provisioner utils states")
```
Comment (ypise) [inline api/python/provisioner/commands/destroy.py:312]: `Teardown Provisioner utils states` to `Teardown foundation states`

File: api/python/provisioner/commands/destroy.py [hunk 45]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
...
+                self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/destroy.py:313]: `utils` to `foundation`

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 46]
```diff
@@ -199,6 +199,11 @@ node {
                 echo "Successfully deployed system states!"
             }
 
+            stage("utils Software Deployment") {
```
Comment (ypise) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:202]: `utils Software Deployment` to `Foundation Deployment`

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 47]
```diff
@@ -199,6 +199,11 @@ node {
...
+                sshCommand remote: remotes[0], command: "provisioner deploy_vm --states utils --setup-type 3_node"
+                echo "Successfully deployed utils states!"
```
Comment (ypise) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:204]: "Successfully deployed utils states!"
to
"Successfully deployed foundation states!"

File: devops/ci/multi_node_deploy/3node_destroy.jenkinsfile [hunk 48]
```diff
@@ -88,6 +88,11 @@ node {
             echo "Successful teardown: 3rd-Party"
         }
 
+        stage("Teardown: utils") {
```
Comment (ypise) [inline devops/ci/multi_node_deploy/3node_destroy.jenkinsfile:91]: "Teardown: utils" to "Teardown: Foundation"

File: devops/ci/multi_node_deploy/3node_destroy.jenkinsfile [hunk 49]
```diff
@@ -88,6 +88,11 @@ node {
...
+            sshCommand remote: remotes[0], command: "provisioner deploy_vm --states utils --setup-type 3_node"
+            echo "Successful teardown: utils"
```
Comment (ypise) [inline devops/ci/multi_node_deploy/3node_destroy.jenkinsfile:93]: "Successful teardown: utils"
to
"Successful teardown: Foundation"

File: devops/ci/single_node_deploy/deploy.jenkinsfile [hunk 50]
```diff
@@ -157,6 +157,11 @@ node {
                 echo "Successfully deployed system states!"
             }
             
+            stage("utils Setup") {
```
Comment (ypise) [inline devops/ci/single_node_deploy/deploy.jenkinsfile:160]: "utils Setup" to "Foundation Setup"

File: devops/ci/single_node_deploy/deploy.jenkinsfile [hunk 51]
```diff
@@ -157,6 +157,11 @@ node {
...
+                sshCommand remote: remote, command: "provisioner deploy_vm --states utils --setup-type single"
+                echo "Successfully deployed utils states!"
```
Comment (ypise) [inline devops/ci/single_node_deploy/deploy.jenkinsfile:162]: "Successfully deployed utils states!"
to
"Successfully deployed foundation states!"

File: devops/ci/single_node_deploy/deploy_iso.jenkinsfile [hunk 52]
```diff
@@ -152,6 +152,13 @@ node {
                 echo "Successfully deployed system states!"
             }
             
+            stage("utils Setup") {
```
Comment (ypise) [inline devops/ci/single_node_deploy/deploy_iso.jenkinsfile:155]: "utils Setup"
to
"Foundation Setup"

File: devops/ci/single_node_deploy/deploy_iso.jenkinsfile [hunk 53]
```diff
@@ -152,6 +152,13 @@ node {
...
+                sshCommand remote: remote, command: """
+                    provisioner deploy_vm --states utils --setup-type single
+                """
+                echo "Successfully deployed utils states!"
```
Comment (ypise) [inline devops/ci/single_node_deploy/deploy_iso.jenkinsfile:159]: "Successfully deployed foundation states!"

File: srv/components/cortx_utils/config.sls [hunk 54]
```diff
@@ -0,0 +1,20 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-utils config:
```
Comment (ypise) [inline srv/components/cortx_utils/config.sls:18]: Missing line:
```  cmd.run:```

`cortx-utils` should be `cortx-py-utils`

File: srv/components/cortx_utils/init_mod.sls [hunk 55]
```diff
@@ -0,0 +1,20 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-utils init:
```
Comment (ypise) [inline srv/components/cortx_utils/init_mod.sls:18]: Missing line:
```
  cmd.run:
```

`cortx-utils` should be `cortx-py-utils`

File: srv/components/cortx_utils/prepare.sls [hunk 56]
```diff
@@ -0,0 +1,20 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-utils prepare:
```
Comment (ypise) [inline srv/components/cortx_utils/prepare.sls:18]: Missing line:
```
  cmd.run:
```

`cortx-utils` should be `cortx-py-utils`

File: srv/components/cortx_utils/sanity_check.sls [hunk 57]
```diff
@@ -0,0 +1,20 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-utils test:
```
Comment (ypise) [inline srv/components/cortx_utils/sanity_check.sls:18]: Missing line:
```
  cmd.run:
```

`cortx-utils` should be `cortx-py-utils`

File: srv/components/cortx_utils/post_install.sls [hunk 58]
```diff
@@ -0,0 +1,21 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-utils post_install:
```
Comment (ypise) [inline srv/components/cortx_utils/post_install.sls:18]: `cortx-utils` should be `cortx-py-utils`

File: api/python/provisioner/commands/deploy_jbod.py [hunk 59]
```diff
@@ -140,6 +144,12 @@ def run(self, **kwargs):
                 self._run_states('system', run_args)
                 self._encrypt_pillar()
 
+            #TODO: is the utils required here for jbod?
+            #if 'utils' in run_args.states:
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_jbod.py:148]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_jbod.py [hunk 60]
```diff
@@ -140,6 +144,12 @@ def run(self, **kwargs):
...
+            #    logger.info("Deploying the utils states")
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_jbod.py:149]: `utils` to `foundation`

File: api/python/provisioner/commands/deploy_jbod.py [hunk 61]
```diff
@@ -140,6 +144,12 @@ def run(self, **kwargs):
...
+            #    self._run_states('utils', run_args)
```
Comment (ypise) [inline api/python/provisioner/commands/deploy_jbod.py:150]: `utils` to `foundation`

**Review by ypise — CHANGES_REQUESTED** [review]

**Review by nitin-seagate — CHANGES_REQUESTED** [review]
@83bhp Please make requested code changes thoroughly and ensure no comments are left unaddressed... We really stable build in today's integration meeting

File: srv/components/cortx_utils/teardown/init.sls [hunk 62]
```diff
@@ -0,0 +1,25 @@
...
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+include:
+  - .reset
+  - .cleanup
+
+Remove cortx-py-utils package:
```
Comment (ypise) [inline srv/components/cortx_utils/teardown/init.sls:22]: `cortx-py-utils` to `cortx-utils`

File: srv/components/cortx_utils/teardown/reset.sls [hunk 63]
```diff
@@ -0,0 +1,20 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Stage - cortx-py-utils Reset:
```
Comment (ypise) [inline srv/components/cortx_utils/teardown/reset.sls:18]: `cortx-py-utils` to `cortx-utils`

**Review by ypise — CHANGES_REQUESTED** [review]

File: devops/ci/single_node_deploy/deploy.jenkinsfile [hunk 64]
```diff
@@ -156,12 +156,17 @@ node {
                 sshCommand remote: remote, command: "provisioner deploy_vm --states system --setup-type single"
                 echo "Successfully deployed system states!"
             }
-            
+
```
Comment (ypise) [inline devops/ci/single_node_deploy/deploy.jenkinsfile:159]: Blank space to be removed.

**Review by ypise — APPROVED** [review]

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile [hunk 65]
```diff
@@ -204,6 +204,11 @@ node {
                 echo "Successfully deployed prereq states!"
             }
 
+            stage("Foundaion Deployment") {
```
Comment (nitin-seagate) [inline devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:207]: Typo - Foundation

File: srv/components/cortx_utils/install.sls [hunk 66]
```diff
@@ -0,0 +1,22 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+Install - cortx-py-utils:
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/install.sls:18]: Please change `cortx-py-utils` to `cortx-utils`

File: srv/components/cortx_utils/prepare.sls [hunk 67]
```diff
@@ -0,0 +1,21 @@
...
+  cmd.run:
+    - name: __slot__:salt:setup_conf.conf_cmd('/opt/seagate/cortx/utils/conf/setup.yaml', 'utils:repare')
```
Comment (nitin-seagate) [inline srv/components/cortx_utils/prepare.sls:20]: Please change `utils:repare` to `utils:prepare`

**Review by nitin-seagate — CHANGES_REQUESTED** [review]
@ypise Please verify the sequence for foundation install or uninstall

File: api/python/provisioner/commands/deploy_jbod.py [hunk 68]
```diff
@@ -140,6 +144,11 @@ def run(self, **kwargs):
                 self._run_states('system', run_args)
                 self._encrypt_pillar()
 
+            if 'utils' in run_args.states:
+               logger.info("Deploying foundation states")
+               self._run_states('utils', run_args)
+               self._encrypt_pillar()
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/deploy_jbod.py:150]: @83bhp    why we don't encrypt in deploy.py and deploy_vm.py?

File: api/python/provisioner/commands/destroy.py [hunk 69]
```diff
@@ -304,6 +308,11 @@ def run(self, **kwargs):  # noqa: C901
                 logger.info("Teardown the system states")
                 self._run_states('system', run_args)
 
+            if 'utils' in run_args.states:
+                logger.info("Teardown foundation states")
+                self._run_states('utils', run_args)
+                self._run_cmd(list_cmds)
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/destroy.py:314]: I guess that should be run only once at the very end but the current code makes it possible to be called few times

**Review by andkononykhin2 — APPROVED** [review]
@83bhp generally it looks good, please just check my comments. Thanks

File: srv/components/cortx_utils/prepare.sls (hunk 67 above)
Comment (83bhp) [reply srv/components/cortx_utils/prepare.sls:20]: done

File: srv/components/cortx_utils/install.sls (hunk 66 above)
Comment (83bhp) [reply srv/components/cortx_utils/install.sls:18]: done

File: devops/ci/multi_node_deploy/3node_deploy.jenkinsfile (hunk 65 above)
Comment (83bhp) [reply devops/ci/multi_node_deploy/3node_deploy.jenkinsfile:212]: done

File: api/python/provisioner/commands/deploy_jbod.py (hunk 68 above)
Comment (83bhp) [reply api/python/provisioner/commands/deploy_jbod.py:150]: good point, but I am really not sure. Will explore soon.

File: api/python/provisioner/commands/destroy.py (hunk 69 above)
Comment (83bhp) [reply api/python/provisioner/commands/destroy.py:314]: thanks for pointing, removed!

**Comment by 83bhp** [comment]
All the comments are resolved, please take a look.

**Pull request merged**